Bluejay.asm

```diff
@@ -1662,95 +1662,186 @@ set_pwm_limit_high_rpm_store:
 
 ;**** **** **** **** **** **** **** **** **** **** **** **** ****
 ;
-; Check motor temperature and limit power
+; Scheduler
 ;
 ;**** **** **** **** **** **** **** **** **** **** **** **** ****
-check_temp_and_limit_power:
-	inc	Adc_Conversion_Cnt			; Increment conversion counter
+
+scheduler_run:
+	; Exit if not 32ms elapsed, otherwise start schedule
+	jbc Flag_32ms_Elapsed, scheduler_start
+	ret
+
+scheduler_start:
+	; Increment Scheduler Counter
+	inc Scheduler_Counter
+
+;******************  128 ms scheduler *******************
+;***********  For PWM temperature limiting **************
+scheduler_128ms_switch_case:
+	; Apply 128ms mask to Scheduler_Counter (2 lowest bits)
+	mov A, Scheduler_Counter
+	anl A, #03h
+
+scheduler_128ms_switch_case_32ms:
+	cjne A, #1, scheduler_128ms_switch_case_64ms
+
+	; Check temp protection enabled, and exit when protection is disabled
+	mov	A, Temp_Prot_Limit
+	jz	scheduler_1024ms_switch_case
+
+	; ******* UPDATE TEMPERATURE SETPOINT *******
+	mov	Temp_Pwm_Level_Setpoint, #255				; Remove setpoint
+
+	; Check TEMP_LIMIT in Base.inc and make calculations to understand temperature readings
+	; Is temperature reading below 256? (ADC 10bit value corresponding to about 25ºC)
+	mov	A, ADC0H									; Load temp hi
+	jz scheduler_1024ms_switch_case					; Temperature below 25ºC do not update setpoint
+
+	mov	A, ADC0L									; Load temp lo
+
 	clr	C
-	mov	A, Adc_Conversion_Cnt		; Is conversion count equal to temp rate?
-	subb	A, #8
-	jc	temp_increase_pwm_limit		; No - increase pwm limit
+	subb	A, Temp_Prot_Limit						; Is temperature below first limit?
+	jc	scheduler_1024ms_switch_case				; Yes - Jump to next scheduler
+
+	mov	Temp_Pwm_Level_Setpoint, #200				; No - update pwm limit (about 80%)
+
+	subb	A, #(TEMP_LIMIT_STEP / 2)				; Is temperature below second limit
```

Comment:
Probably a stupid question: division by two is the same as right shifting the value once - or will this part be assembled into something different?

Reply:
That division is handled by the Keil assemby preprocessor

Reply:
Ah yeah, makes sense. TEMP_LIMIT_STEP does not change during execution...